docs/sample_config.yaml

```diff
@@ -1424,6 +1424,99 @@ saml2_config:
+    #
+    #discover: true
+
+    # the OIDC issuer. Used to validate tokens and discover the providers endpoints. Required.
+    #
+    #issuer: "https://accounts.example.com/"
+
+    # oauth2 client id to use. Required.
+    #
+    #client_id: "provided-by-your-issuer"
+
+    # oauth2 client secret to use. Required.
+    #
+    #client_secret: "provided-by-your-issuer"
+
+    # auth method to use when exchanging the token.
+    # Valid values are "client_secret_basic" (default), "client_secret_post" and "none".
+    #
+    #client_auth_method: "client_auth_basic"
+
+    # list of scopes to ask. This should include the "openid" scope. Defaults to ["openid"].
+    #
+    #scopes: ["openid"]
+
+    # the oauth2 authorization endpoint. Required if provider discovery is disabled.
+    #
+    #authorization_endpoint: "https://accounts.example.com/oauth2/auth"
+
+    # the oauth2 token endpoint. Required if provider discovery is disabled.
+    #
+    #token_endpoint: "https://accounts.example.com/oauth2/token"
+
+    # the OIDC userinfo endpoint. Required if discovery is disabled and the "openid" scope is not asked.
+    #
+    #userinfo_endpoint: "https://accounts.example.com/userinfo"
+
+    # URI where to fetch the JWKS. Required if discovery is disabled and the "openid" scope is used.
+    #
+    #jwks_uri: "https://accounts.example.com/.well-known/jwks.json"
+
+    # skip metadata verification. Defaults to false.
+    # Use this if you are connecting to a provider that is not OpenID Connect compliant.
+    # Avoid this in production.
+    #
+    #skip_verification: false
+
+    # name of the claim containing a unique identifier for the user.
+    # OpenID Connect compliant providers should provide a `sub` claim for that.
+    #
+    #subject_claim: "sub"
+
+    # defines how the user info from the OIDC provider are mapped to user properties.
+    # Those are Jinja2 templates, where the userinfo object is available through the `user` variable.
+    #
+    mapping_templates:
+        # the localpart of the MXID. Defaults to '{{ user.preferred_username }}'.
+        #
+        #localpart: '{{ user.preferred_username }}'
+
+        # display name to set on first login. Defaults to '{{ user.given_name }} {{ user.last_name }}'.
+        #
+        #display_name: '{{ user.given_name }} {{ user.last_name }}'
+
+    # Directory in which Synapse will try to find the template files below.
+    # If not set, default templates from within the Synapse package will be used.
+    #
+    # DO NOT UNCOMMENT THIS SETTING unless you want to customise the templates.
+    # If you *do* uncomment it, you will need to make sure that all the templates
+    # below are in the directory.
+    #
+    # Synapse will look for the following templates in this directory:
+    #
+    # * HTML page to display to users if something goes wrong during the
+    #   authentication process: 'oidc_error.html'.
```

Comment:
Does this differ dramatically from the `saml_error.html` page? It'd be nice if this used the `sso.template_dir` setting instead of adding another one.